### PR TITLE
EventHub : Added Examples and patch, check name availability operations and RP operations 

### PR DESCRIPTION
**Pull request by v-Ajnava**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [X] The title of the PR is clear and informative.
- [X] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [X] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [X] If applicable, the PR references the bug/issue that it fixes.
- [X] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [X] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [X] My spec meets the review criteria:
  - [X] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [X] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [X] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Review by olydis — COMMENTED** [review]
@v-Ajnava Thanks for providing the examples! Your changes to the Swagger look good (:+1: for extracting reused parameters to the top level).

However, there are some "low hanging fruits" that are unrelated to your changes: Some descriptions have low quality. They are used for producing documentation and hence directly affect the customer experience. For example, I see descriptions saying `Description of ...` which provides absolutely no value (imagine looking for documentation and then finding this - instead of an actual description! I'd expect a text describing what the object *represents*). Spelling is also inconsistent here and there ("Event Hub", "eventhub").

Please make a quick pass over the descriptions and this is good to go!

**Comment by v-Ajnava** [comment]
@olydis, updated descriptions, please review and let me know your comments.

**Review by olydis — APPROVED** [review]

File: arm-eventhub/2015-08-01/swagger/EventHub.json [hunk 1]
```diff
@@ -1751,7 +1744,69 @@
...
+        "NameInLockdown",
+        "TooManyNamespaceInCurrentSubscription"
+      ],
+      "x-ms-enum": {
+        "name": "UnavailableReason",
+        "modelAsString": false
+      },
+      "description": "Specifies the reason for the unavailability of the service."
+    },
+    "CheckNameAvailability": {
```
Comment (olydis) [inline arm-eventhub/2015-08-01/swagger/EventHub.json]: is this name intended? Usually you'd call this "...Parameter(s)" or similar

File: arm-eventhub/2015-08-01/swagger/EventHub.json [hunk 2]
```diff
@@ -1751,7 +1744,69 @@
...
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "The Name to check the namespce name availability"
+        }
+      },
+      "required": [
+        "name"
+      ],
+      "description": "Check Name availability of Namespace request properties."
```
Comment (olydis) [inline arm-eventhub/2015-08-01/swagger/EventHub.json]: spelling: "Name"

File: arm-eventhub/2015-08-01/swagger/EventHub.json [hunk 3]
```diff
@@ -1751,7 +1744,69 @@
...
+          "$ref": "#/definitions/UnavailableReason",
+          "description": "The reason for unavailability of a Namespace."
+        },
+        "message": {
+          "type": "string",
+          "readOnly": true,
+          "description": "The detailed info regarding the reason associated with the Namespace."
+        }
+      },
+      "description": "Check Name availability of Namespace response properties."
```
Comment (olydis) [inline arm-eventhub/2015-08-01/swagger/EventHub.json]: same description and issue as before

File: arm-eventhub/2015-08-01/swagger/EventHub.json [hunk 4]
```diff
@@ -1768,6 +1823,56 @@
...
+    },
+    "authorizationRuleNameParameter": {
+      "name": "authorizationRuleName",
+      "in": "path",
+      "required": true,
+      "type": "string",
+      "minLength": 1,
+      "maxLength": 50,
+      "x-ms-parameter-location": "method",
+      "description": "The authorizationrule name."
```
Comment (olydis) [inline arm-eventhub/2015-08-01/swagger/EventHub.json]: spelling intended?

File: arm-eventhub/2015-08-01/swagger/EventHub.json [hunk 5]
```diff
@@ -1768,6 +1823,56 @@
...
+    },
+    "resourceGroupNameParameter": {
+      "name": "resourceGroupName",
+      "in": "path",
+      "required": true,
+      "type": "string",
+      "minLength": 1,
+      "maxLength": 90,
+      "x-ms-parameter-location": "method",
+      "description": "Name of the Resource group within the Azure subscription."
```
Comment (olydis) [inline arm-eventhub/2015-08-01/swagger/EventHub.json]: capitalization intended?

**Review by olydis — COMMENTED** [review]
nits

File: arm-eventhub/2015-08-01/swagger/EventHub.json (hunk 4 above)
Comment (v-Ajnava) [inline arm-eventhub/2015-08-01/swagger/EventHub.json]: Corrected.

File: arm-eventhub/2015-08-01/swagger/EventHub.json (hunk 5 above)
Comment (v-Ajnava) [inline arm-eventhub/2015-08-01/swagger/EventHub.json]: Corrected.

File: arm-eventhub/2015-08-01/swagger/EventHub.json (hunk 3 above)
Comment (v-Ajnava) [inline arm-eventhub/2015-08-01/swagger/EventHub.json]: Description updated.

File: arm-eventhub/2015-08-01/swagger/EventHub.json (hunk 2 above)
Comment (v-Ajnava) [inline arm-eventhub/2015-08-01/swagger/EventHub.json]: Description changed

File: arm-eventhub/2015-08-01/swagger/EventHub.json (hunk 1 above)
Comment (v-Ajnava) [inline arm-eventhub/2015-08-01/swagger/EventHub.json]: updated with 'Parameters'

File: arm-eventhub/2015-08-01/swagger/EventHub.json [hunk 6]
```diff
@@ -1762,17 +1762,17 @@
       },
       "description": "Specifies the reason for the unavailability of the service."
     },
-    "CheckNameAvailability": {
+    "CheckNameAvailabilityParameter": {
       "properties": {
         "name": {
           "type": "string",
-          "description": "The Name to check the namespce name availability"
+          "description": "Name to check the namespace name availability"
```
Comment (olydis) [inline arm-eventhub/2015-08-01/swagger/EventHub.json]: good catch

**Pull request merged**